tv_apps/browser/js/awesomescreen.js

```diff
@@ -3266,5 +3306,219 @@ var Awesomescreen = {
         .querySelector('.tab-content[data-content="' + dataContent +'"]')
         .classList.add('show');
     }
+  },
+
+  initSpatialNavigation: function awesomescreen_initSpatialNavigation() {
+    SpatialNavigation.init();
+
+    this.enableSpatialNavigation();
+
+    SpatialNavigation.makeFocusable();
+
+    SpatialNavigation.focus('tool-bar');
+
+    document.querySelector('.tab-navigation').addEventListener('sn:willmove',
+      function(e){
+        let target = e.target;
+        if (!target.classList.contains('tab-option')) {
+          return;
+        }
+
+        let detail = e.detail;
+        if (detail.direction === 'left' && !target.previousElementSibling) {
+          e.preventDefault();
+        }
+        if (detail.direction === 'right' && !target.nextElementSibling) {
+          e.preventDefault();
+        }
+      });
+
+    document.querySelector('#sync-tab-content .tab-list-view')
+      .addEventListener('keydown', function(e){
+        let firsListItem =
+          document
+            .querySelector(`#sync-tab-content
+              .tab-list-item:first-child .list-item`);
+
+        if (firsListItem !== e.target || e.keyCode !== KeyEvent.DOM_VK_UP) {
+          e.stopPropagation();
+        }
+      });
+  },
+
+  enableSpatialNavigation: function awesomescreen_disableSpatialNavigation() {
+    if (this.isSpatialNavigationEnable) {
+      return;
+    }
+
+    function getSearchBarStyle() {
```

Comment:
I'd prefer `checkIsSearch` or something like this. The "get" sounds like to retrieve something.

However, since this function is only used once, do we really need a function here?